app/frontend/helpers/epochBoundaryUtils.ts

```diff
@@ -0,0 +1,21 @@
+import * as moment from 'moment'
+import {InternalError, InternalErrorReason} from '../errors'
+import {OptionalParams} from '../errors/BaseError'
+
+export function isEpochBoundaryUnderway(): boolean {
+  const epochTransition = moment('15 May 2021 21:44 UTC')
+  const now = moment(Date.now())
+  const dayDiff = now.diff(epochTransition, 'days')
+  const relativeMinuteDiff = now.subtract(dayDiff, 'days').diff(epochTransition, 'minutes')
+  // multiple of 5 days passed and it's no more than X minutes since the transition
+  if (dayDiff % 5 === 0 && relativeMinuteDiff < 60) {
+    return true
+  }
+  return false
+}
+
+export function probeEpochBoundary(params?: OptionalParams): void {
```

Comment:
I'd rename it `throwIfEpochBoundary` to make it clear it throws an error and also to make it easier to follow when used together with throwing other errors in the logic that calls it